src/ameba/formatter/github_actions_formatter.cr

```diff
@@ -37,6 +43,133 @@ module Ameba::Formatter
       end
     end
 
+    # Reports a message when inspection is finished.
+    def finished(sources) : Nil
+      return unless step_summary_file = ENV["GITHUB_STEP_SUMMARY"]?
+
+      if started_at = @started_at
+        time_elapsed = Time.monotonic - started_at
+      end
+
+      File.write(step_summary_file, summary(sources, time_elapsed))
+    end
+
+    private def summary(sources, time_elapsed)
+      failed_sources = sources.reject(&.valid?)
+      total = sources.size
+      failures = failed_sources.sum(&.issues.count(&.enabled?))
+
+      String.build do |output|
+        output << "## Ameba Results %s\n\n" % {
+          failures == 0 ? ":green_heart:" : ":bug:",
+        }
+
+        if failures.positive?
+          output << "### Issues found:\n\n"
+
+          failed_sources.each do |source|
+            issue_count = source.issues.count(&.enabled?)
+
+            if issue_count.positive?
+              output << "#### `%s` (**%d** %s)\n\n" % {
+                source.path,
+                issue_count,
+                pluralize(issue_count, "issue"),
+              }
+
+              output.puts "| Line | Severity | Name | Message |"
+              output.puts "| ---- | -------- | ---- | ------- |"
+
+              source.issues.each do |issue|
+                next if issue.disabled?
+
+                output.puts "| %s | %s | %s | %s |" % {
+                  issue_location_value(issue) || "-",
+                  issue.rule.severity,
+                  issue.rule.name,
```

Comment:
**suggestion**: Maybe the rule name could include a link to the documentation?

Reply:
I thought about it, although atm we could only link directly to the ameba repo, since we don't have versioned docs (yet).

See related discussion in [vscode-crystal-ameba](https://github.com/crystal-ameba/vscode-crystal-ameba) repo: https://github.com/crystal-ameba/vscode-crystal-ameba/pull/146#discussion_r1845516181